examples-standalone/personal-finance-app/src/app/components/navigation/navigation.component.ts

```diff
@@ -0,0 +1,121 @@
+import { Component, OnInit, ViewEncapsulation } from '@angular/core';
+import { NavigationEnd, Router, RouterModule } from '@angular/router';
+import { KENDO_ICONS } from '@progress/kendo-angular-icons';
+import { MessageService } from '@progress/kendo-angular-l10n';
+import {
+  DrawerSelectEvent,
+  KENDO_LAYOUT,
+} from '@progress/kendo-angular-layout';
+import {
+  arrowsSwapIcon,
+  chartColumnStackedIcon,
+  dollarIcon,
+  gearIcon,
+  gridIcon,
+  menuIcon,
+  sparklesIcon,
+  SVGIcon,
+} from '@progress/kendo-svg-icons';
+import { DrawerItem } from '../../models/drawer-item';
+import { CustomMessagesService } from '../../services/custom-messages.service';
+import { FooterComponent } from '../footer/footer.component';
+import { HeaderComponent } from '../header/header.component';
+import { filter } from 'rxjs';
+
+@Component({
+  selector: 'app-navigation',
+  standalone: true,
+  imports: [
+    RouterModule,
+    KENDO_LAYOUT,
+    KENDO_ICONS,
+    HeaderComponent,
+    FooterComponent,
+  ],
+  templateUrl: './navigation.component.html',
+  styleUrl: './navigation.component.css',
+  encapsulation: ViewEncapsulation.None,
+})
+export class NavigationComponent implements OnInit {
+  public items: DrawerItem[] = [];
+  public menuIcon: SVGIcon = menuIcon;
+  public expanded: boolean = false;
+
+  public customMsgService: CustomMessagesService;
+
+  private selectedItem: string = '';
+
+  ngOnInit() {
+    this.router.events
+      .pipe(filter((event) => event instanceof NavigationEnd))
+      .subscribe((event: NavigationEnd) => {
+        if (event.url === '/') {
+          this.selectedItem = '';
+        } else {
+          this.selectedItem = event.url;
+        }
+        this.items = this.drawerItems(this.selectedItem);
+      });
+  }
+
+  constructor(private messages: MessageService, private router: Router) {
+    this.customMsgService = this.messages as CustomMessagesService;
+
+    this.items = this.drawerItems(this.selectedItem);
+
+    this.customMsgService.localeChange.subscribe(() => {
+      this.items = this.drawerItems(this.selectedItem);
+    });
+  }
+
+  public onItemSelection(ev: DrawerSelectEvent) {
```

Comment:
Additionally, I suggest keeping parameter names consistent when handling events. Some components use `e`, while others use `ev`. The naming should be consistent across the application.
```suggestion
  public onItemSelection(ev: DrawerSelectEvent): void {
```